FogCreek/Glitch-Community
# Oceanic ornament 1: preventing tabs from outside of the new stuff dialog

## Links
* https://oceanic-ornament-1.glitch.me/
* https://glitch.manuscript.com/f/cases/3328202/

## Changes:
* Previously when you opened the "new stuff" pupdate and you hit tab you'd tab through the modal, but then you'd also tab through the other elements behind the modal. This pr makes it so that when the new stuff update modal is open you tab through all of the focusable elements available in a circle and to exit you hit escape. 

## How To Test:
* you can open the pupdate by logging in and clicking on your avatar on the upper right hand side and clicking "new stuff" in the dropdown, then try to tab around, at the moment you should be stuck on the checkbox as that's the only focusable element there. but if you add another button you should see that it's tab-able. to exit hit escape. 

## Feedback I'm looking for:
- does this functionality seem right? I implemented it because that's what was outlined in the case and in [this article](https://bitsofco.de/accessible-modal-dialog/) and sounded right but as I play around with it I wonder if instead of going in a circle of focusable elements it should instead auto-close the modal and return your focus to the pup once you've gone through all the focusable elements? I can see an argument for a circle if it had some kind of required acknowledgement in it like a terms of service, but in this case maybe this is making it more difficult to use? what do we think?
- do we have other dialog boxes like this where we'd want this kind of functionality? i did a brief search and it seemed like no?
- am I doing anything funky with useEffect/useRef/useState? I'm still getting a feel for them. 
- I could be wrong but in my testing it seemed like the other accessibility issues with the new stuff dialog outlined in the cases had been fixed already, but definitely let me know if I missed anything!

## Things left to do before deploying:
- [x] try chrome
- [x] test firefox
- [x] test safari
- [x] test edge
